src/index.ts

```diff
@@ -149,36 +149,18 @@ class ClientApp {
         pcOrigin?: string;
     } = {}) {
         if (cfg) {
-            if (cfg.hasOwnProperty('pcEnvironmentQueryParam')) {
+            if ('pcEnvironmentQueryParam' in cfg) {
                 const paramName = cfg.pcEnvironmentQueryParam;
-
-                if (typeof paramName !== 'string' || paramName.trim().length === 0) {
-                    throw new Error('Invalid query param name provided.  Must be non-null, non-empty string');
-                }
-
+                this.assertNonEmptyString(paramName, 'query param name');
                 const parsedQueryString = queryString.parse(ClientApp._getQueryString() || '');
                 const paramValue = parsedQueryString[paramName];
-                if (paramValue && typeof paramValue === 'string') {
-                    this._pcEnv = envUtils.lookupPcEnv(paramValue, true);
-                    if (!this._pcEnv) {
-                        throw new Error(`Could not parse '${paramValue}' into a known PureCloud environment`);
-                    }
-                } else {
-                    throw new Error(`Could not find unique value for ${paramName} parameter on Query String`);
-                }
-            } else if (cfg.hasOwnProperty('pcEnvironment')) {
-                if (typeof cfg.pcEnvironment !== 'string' || cfg.pcEnvironment.trim().length === 0) {
-                    throw new Error('Invalid pcEnvironment provided.  Must be a non-null, non-empty string');
-                }
-                this._pcEnv = envUtils.lookupPcEnv(cfg.pcEnvironment, true);
-                if (!this._pcEnv) {
-                    throw new Error(`Could not parse '${cfg.pcEnvironment}' into a known PureCloud environment`);
-                }
-            } else if (cfg.hasOwnProperty('pcOrigin')) {
-                if (typeof cfg.pcOrigin !== 'string' || cfg.pcOrigin.trim().length === 0) {
-                    throw new Error('Invalid pcOrigin provided.  Must be a non-null, non-empty string');
-                }
-
+                this.assertNonEmptyString(paramValue, 'query param value');
+                this._pcEnv = this.lookupEnv(paramValue);
```

Comment:
I think this creates a change in behavior, right?

As lookupEnv doesn't throw, it will be null and then overridden by the default?  I think what we had (and still want?) is to fail out if the value provided isn't valid, right?

Reply:
ugh ... I'm an idiot.  nvmd